rsconf/pkcli/setup_dev.py

```diff
@@ -127,7 +127,12 @@ def _add_host(j2_ctx, srv, host):
     from rsconf.component import docker_registry
     from rsconf.component import rsconf
 
-    netrc = rsconf.host_init(j2_ctx, host)
-    pkio.write_text(srv.join(host + "-netrc"), netrc)
+    def _netrc():
+        for l in rsconf.host_init(j2_ctx, host).split("(?<=\n)"):
+            if l.startswith("machine"):
+                return l
+        raise AssertionError("format of host_init changed no machine line found")
```

Comment:
Getting this on a fresh build
```
rsconf (463-opendkim)$ rm -rf run
rsconf (463-opendkim)$ rsconf build
Traceback (most recent call last):
  File "/home/vagrant/.pyenv/versions/py3/bin/rsconf", line 8, in <module>
    sys.exit(main())
  File "/home/vagrant/src/radiasoft/rsconf/rsconf/rsconf_console.py", line 17, in main
    return pkcli.main("rsconf")
  File "/home/vagrant/src/radiasoft/pykern/pykern/pkcli/__init__.py", line 166, in main
    res = argh.dispatch(parser, argv=argv)
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/argh/dispatching.py", line 179, in dispatch
    return run_endpoint_function(
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/argh/dispatching.py", line 237, in run_endpoint_function
    return _process_command_output(lines, output_file, raw_output)
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/argh/dispatching.py", line 254, in _process_command_output
    for line in lines:
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/argh/dispatching.py", line 362, in _execute_command
    for line in result:
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/argh/dispatching.py", line 342, in _call
    result = function(*positional, **keywords)
  File "/home/vagrant/src/radiasoft/rsconf/rsconf/pkcli/build.py", line 136, in default_command
    setup_dev.default_command()
  File "/home/vagrant/src/radiasoft/rsconf/rsconf/pkcli/setup_dev.py", line 71, in default_command
    _add_host(j2_ctx, srv, h)
  File "/home/vagrant/src/radiasoft/rsconf/rsconf/pkcli/setup_dev.py", line 136, in _add_host
    pkio.write_text(srv.join(host + "-netrc"), _netrc())
  File "/home/vagrant/src/radiasoft/rsconf/rsconf/pkcli/setup_dev.py", line 134, in _netrc
    raise AssertionError("format of host_init changed no machine line found")
AssertionError: format of host_init changed no machine line found
```

Reply:
f3fb31d try now

Reply:
```
rsconf (463-opendkim)$ rm -rf run
rsconf (463-opendkim)$ rsconf build
rsconf/pkcli/build.py:83:require_component opendkim: component failed:
rsconf/pkcli/build.py:65:create_host v3.radia.run: host failed:
Traceback (most recent call last):
  File "/home/vagrant/.pyenv/versions/py3/bin/rsconf", line 8, in <module>
    sys.exit(main())
  File "/home/vagrant/src/radiasoft/rsconf/rsconf/rsconf_console.py", line 17, in main
    return pkcli.main("rsconf")
  File "/home/vagrant/src/radiasoft/pykern/pykern/pkcli/__init__.py", line 166, in main
    res = argh.dispatch(parser, argv=argv)
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/argh/dispatching.py", line 179, in dispatch
    return run_endpoint_function(
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/argh/dispatching.py", line 237, in run_endpoint_function
    return _process_command_output(lines, output_file, raw_output)
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/argh/dispatching.py", line 254, in _process_command_output
    for line in lines:
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/argh/dispatching.py", line 362, in _execute_command
    for line in result:
  File "/home/vagrant/.pyenv/versions/3.9.15/envs/py3/lib/python3.9/site-packages/argh/dispatching.py", line 342, in _call
    result = function(*positional, **keywords)
  File "/home/vagrant/src/radiasoft/rsconf/rsconf/pkcli/build.py", line 154, in default_command
    t.create_host(new_d)
  File "/home/vagrant/src/radiasoft/rsconf/rsconf/pkcli/build.py", line 57, in create_host
    self.require_component(*self.hdb.rsconf_db.components)
  File "/home/vagrant/src/radiasoft/rsconf/rsconf/pkcli/build.py", line 81, in require_component
    self.build_component(c)
  File "/home/vagrant/src/radiasoft/rsconf/rsconf/pkcli/build.py", line 45, in build_component
    compt.build_compile()
  File "/home/vagrant/src/radiasoft/rsconf/rsconf/component/__init__.py", line 76, in build_compile
    self.internal_build_compile()
  File "/home/vagrant/src/radiasoft/rsconf/rsconf/component/opendkim.py", line 20, in internal_build_compile
    if self._named_compile(jc, z):
  File "/home/vagrant/src/radiasoft/rsconf/rsconf/component/opendkim.py", line 150, in _named_compile
    self._read_keys(jc, z)
  File "/home/vagrant/src/radiasoft/rsconf/rsconf/component/opendkim.py", line 124, in _read_keys
    v.rows.append(opendkim.gen_key(v.secret_d, d))
  File "/home/vagrant/src/radiasoft/rsconf/rsconf/pkcli/opendkim.py", line 28, in gen_key
    subprocess.check_call(
  File "/home/vagrant/.pyenv/versions/3.9.15/lib/python3.9/subprocess.py", line 368, in check_call
    retcode = call(*popenargs, **kwargs)
  File "/home/vagrant/.pyenv/versions/3.9.15/lib/python3.9/subprocess.py", line 349, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/home/vagrant/.pyenv/versions/3.9.15/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/home/vagrant/.pyenv/versions/3.9.15/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'opendkim-genkey'
```

Reply:
For now dnf install opendkim. We don't have a solution to https://github.com/radiasoft/rsconf/issues/173